internal/adapters/operation/operation_flow_redis.go

```diff
@@ -80,37 +80,54 @@ func (r *redisOperationFlow) InsertPriorityOperationID(ctx context.Context, sche
 
 // NextOperationID fetches the next scheduler operation ID from the
 // pending_operations list.
-func (r *redisOperationFlow) NextOperationID(ctx context.Context, schedulerName string) (opId string, err error) {
-	var opIDs []string
-	metrics.RunWithMetrics(operationFlowStorageMetricLabel, func() error {
-		opIDs, err = r.client.BLPop(ctx, 0, r.buildSchedulerPendingOperationsKey(schedulerName)).Result()
-		return err
-	})
-	if err != nil {
-		return "", errors.NewErrUnexpected("failed to fetch next operation ID").WithError(err)
+func (r *redisOperationFlow) NextOperationID(ctx context.Context, schedulerName string) (opID string, err error) {
+	opID, err = r.fetchNextOpIDFromAuxiliaryQueue(ctx, schedulerName)
+
+	if err == nil {
+		return opID, nil
 	}
 
-	if len(opIDs) == 0 {
-		return "", errors.NewErrNotFound("scheduler \"%s\" has no operations", schedulerName)
+	if err != nil && err != redis.Nil {
+		return "", errors.NewErrUnexpected("failed to fetch next operation ID from auxiliary queue").WithError(err)
 	}
 
-	// Once redis finds any operation ID, it returns an array using:
-	// - the first position to indicate the list name;
-	// - the second position with the found value.
-	opId = opIDs[1]
-	return opId, nil
+	opID, err = r.waitAndFetchNextOpIDFromMainQueue(ctx, schedulerName)
+
+	if err != nil {
+		return "", errors.NewErrUnexpected("failed to fetch next operation ID from main queue").WithError(err)
+	}
+
+	return opID, nil
 }
 
-func (r *redisOperationFlow) ListSchedulerPendingOperationIDs(ctx context.Context, schedulerName string) (operationsIDs []string, err error) {
+// RemoveNextOperation removes the next operation from the operation flow.
+func (r *redisOperationFlow) RemoveNextOperation(ctx context.Context, schedulerName string) (err error) {
```

Comment:
What do you think to rename this function to something like: `FinishOperation` since it is removing the operation from the `auxiliary` queue and in the practice, it is finishing the operation to the Flow struct.

Reply:
We already have a FinishOperation method in OperationsManager, besides that, this method does not "finishes" the operation, it removes the operation from what we call "flow". The operation is finished when it finishes getting executed.